### PR TITLE
Additional fixes for JSX transforms

### PR DESCRIPTION
**Pull request by rbuckton**
This fixes qualified names for JSX emit, formatting for TSX emit, and a few other odds and ends with the JSX transforms.

### CODE REVIEW

**Comment by rbuckton** [comment]
CC: @vladima, @DanielRosenwasser, @yuit

File: src/compiler/printer.ts [hunk 1]
```diff
@@ -1536,7 +1534,7 @@ const _super = (function (geti, seti) {
                 write("interface ");
                 emit(node.name);
                 emitTypeParameters(node, node.typeParameters);
-                emitList(node, node.heritageClauses, ListFormat.SingleLine);
+                emitList(node, node.heritageClauses, ListFormat.HeritageClauses);
```
Comment (RyanCavanaugh) [inline src/compiler/printer.ts]: This seems unrelated to JSX?

File: tests/baselines/reference/tsxReactEmitEntities.js [hunk 2]
```diff
@@ -12,5 +12,5 @@ declare var React: any;
 
 
 //// [file.js]
-React.createElement("div", null, "Dot goes here: · &notAnEntity; ");
+React.createElement("div", null, "Dot goes here: \u00B7 &notAnEntity; ");
```
Comment (RyanCavanaugh) [inline tests/baselines/reference/tsxReactEmitEntities.js]: I'm not entirely sure if it matters, but this behavior is different from Babel's transformer. In theory we want a dot character here instead of a \u sequence.

**Comment by RyanCavanaugh** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
LGTM so far

File: tests/baselines/reference/tsxReactEmitEntities.js (hunk 2 above)
Comment (rbuckton) [inline tests/baselines/reference/tsxReactEmitEntities.js]: This is because the synthetic literal now passes through `escapeNonAsciiCharacters` which encodes the dot.

Comment (rbuckton) [inline tests/baselines/reference/tsxReactEmitEntities.js]: I will accept this baseline for now, but may revisit as we discuss whether we need to call `escapeNonAsciiCharacters` for every synthetic literal. It may only really be useful for performing string canonicalization for module import deduplication.
